lib/blacklight/models/assessment.rb

```diff
@@ -0,0 +1,95 @@
+require "blacklight/models/assignment_group"
+require "blacklight/models/assignment"
+require "blacklight/models/question"
+
+module Blacklight
+  class Assessment
+    def initialize
+      @title = ""
+      @description = ""
+      @quiz_type = "assignment"
+      @points_possible = 0
+      @items = []
+      @group_name
+      @group_id
+      @workflow_state = "published"
+      @available = true
+    end
+
+    def iterate_xml(data)
+      @title = data.children.at("assessment").attributes["title"].value
+      @points_possible = data.children.at("qmd_absolutescore_max").text
+      @description = data.children.at("presentation_material").
+        children.at("mat_formattedtext").text
+      @group_name = data.children.at("bbmd_assessmenttype").text
+      data.children.at("section").children.map do |item|
+        @items.push(item) if item.name == "item"
+      end
+      self
+    end
+
+    def canvas_conversion(course)
+      if @items.count.positive?
+        assessment = CanvasCc::CanvasCC::Models::Assessment.new
+        assessment.identifier = Blacklight.create_random_hex
+        course = create_assignment_group(course)
+        assignment = create_assignment
+        assignment.quiz_identifier_ref = assessment.identifier
+        course.assignments << assignment
+        assessment = setup_assessment(assessment, assignment)
+        course.assessments << assessment
+      end
+      course
+    end
+
+    def setup_assessment(assessment, assignment)
+      assessment.title = @title
+      assessment.description = @description
+      assessment.available = @available
+      assessment.quiz_type = @quiz_type
+      assessment.points_possible = @points_possible
+      assessment = create_items(assessment)
+      assessment.assignment = assignment
+      assessment
+    end
+
+    def create_items(assessment)
+      items = []
+      @items = @items - ["", nil]
+      @items.each do |item|
```

Comment:
I think it makes sense to keep `item` out of `Question`, since `item`shouldn't be part of the question's persistent state, but it might make sense to have `Question.from` both instantiate the question, _and_ invoke `#iterate_xml` (in other words, have `Question.from` return the fully initialized Question object).